src/olympia/reviews/views.py

```diff
@@ -327,22 +327,46 @@ class ReviewViewSet(AddonChildMixin, ModelViewSet):
 
     queryset = Review.objects.all()
 
+    def set_addon_object_from_review(self, review):
+        """Set addon object on the instance from a review object."""
+        self.kwargs['addon_pk'] = str(review.addon.pk)
+        del self.addon_object  # Force get_addon_object() to try again.
+        return self.get_addon_object()
```

Comment:
I feel like the comment is misguiding, maybe `Force get_addon_object to refresh self.addon_object with the new add-on`?